File: BitsHelper/BitsHelper/ModFolder/Loenn/entities/alterEgoBlockField.lua
local entity = {}

entity.name = "BitsHelper/AlterEgoBlockField"
entity.fillColor = {0.2, 0.4, 0.6, 0.4}
entity.borderColor = {1.0, 0.4, 1.0, 1.0}
entity.placements = {
    {
        name = "in",
        data = {
            width = 8,
            height = 8,
            blockIn = true,
            blockOut = false
        }
    },
    {
        name = "out",
        data = {
            width = 8,
            height = 8,
            blockIn = false,
            blockOut = true
        }
    }
}

return entity

File: BitsHelper/BitsHelper/Entities/AlterEgoBlockField.cs
using Celeste.Mod.Entities;

namespace Celeste.Mod.BitsHelper.Entities;

[CustomEntity("BitsHelper/AlterEgoBlockField"), Tracked]
public sealed class AlterEgoBlockField : Entity
{
    private readonly bool blockIn, blockOut;

    public bool BlockIn => blockIn;

    public bool BlockOut => blockOut;

    public AlterEgoBlockField(EntityData data, Vector2 offset)
        : this(data.Position + offset, data.Width, data.Height, data.Bool("blockIn", false), data.Bool("blockOut", false))
    {
    }

    public AlterEgoBlockField(Vector2 position, int width, int height, bool blockIn, bool blockOut)
        : base(position)
    {
        Collider = new Hitbox(width, height);
        this.blockIn = blockIn;
        this.blockOut = blockOut;
    }
}
